packages/vinxi/lib/dev-server.js

```diff
@@ -8,7 +8,7 @@ import {
 } from "h3";
 import { createNitro } from "nitropack";
 
-import { join } from "node:path";
+import { join } from "pathe";
```

Comment:
Same thing here as in build.js.

https://github.com/nksaraf/vinxi/blob/505dd51fd130dc6747de2ff5259ef0cf86f62b6f/packages/vinxi/lib/dev-server.js#L156-L160

Reply:
So, using `pathe` helps in these Windows situations? if yes, I am so in with all these changes. I have a tough time maintaining windows coz I don't have access to a machine

Reply:
At the very least it should not cause problems for linux since `pathe` uses `/`.
The only change I am unsure about is `path: join(router.base, "@fs", absolutePath)` on `dev-server-manifest.js`. 

From a quick test it looks like as long things go through vite/rollup it's fine to import with `C:/some/path`

```js
import test from "C:/Users/Davide/Desktop/wpath/w/dep.js"
```

Node:  :x: not supported
Vite/rollup: :heavy_check_mark:

```js
import test from "C:\\Users\\Davide\\Desktop\\wpath\\w\\dep.js"
```

Node: :x: not supported
Vite/rollup: :heavy_check_mark: 


```js
import test from "file:///C:/Users/Davide/Desktop/wpath/w/dep.js"
```

Node: :heavy_check_mark:
Vite/rollup: :x: rollup can't resolve

```js
import test from "/C:/Users/Davide/Desktop/wpath/w/dep.js"
```

Node: :heavy_check_mark:
Vite/rollup: :heavy_check_mark:

Reply:
So you have tested these changes on windows? and they are all working good? is there anything else that's not working on windows?

Reply:
TBH my brain hurts when I try to decipher this windows/linux path differences, I never know what to do so this a big saving grace for me if you have got it all working

Reply:
It's honestly quite difficult to tell if verything actually works. I am fighting with a lot of inconsistency. Primarly used the tests here and the bare example on the solid-start/vinxi branch. The example on the solid-start repo is not reliable as it had other issues unrelated to vinxi.

If there is anything specific you want me to run on my machine let me know.

I just run the tests `main` and `windows` branch on a code space. One test just randomly fails if I run it enough times. It also looks like it's not releasing the ports?

Windows `windows` branch: two additional tests fail (on `main` none pass). Hmm...
```shell
  3 failed
    [chromium] › basic-dev.test.ts:40:3 › basic dev › api ==========================================
    [chromium] › basic-prod.test.ts:40:3 › basic build › api =======================================
    [chromium] › hmr.test.ts:81:3 › hmr › hmr api ==================================================
  6 passed (39s)
```

Linux `windows` branch:

```shell
  1) [chromium] › hmr.test.ts:81:3 › hmr › hmr api =================================================

    Error: expect(received).toBe(expected) // Object.is equality

    - Expected  - 1
    + Received  + 7

    - Hello world
    + <!DOCTYPE html><html lang="en"><head><link rel="icon" href="/favicon.ico"/><!--$--><script type="module">
    + import RefreshRuntime from "/_build/@react-refresh"
    + RefreshRuntime.injectIntoGlobalHook(window)
    + window.$RefreshReg$ = () => {}
    + window.$RefreshSig$ = () => (type) => type
    + window.__vite_plugin_react_preamble_installed__ = true
    + </script><script type="module" src="/_build/@vite/client"></script><!--/$--></head><body><section><h1 data-test-id="content">Hello from Vinxi</h1><div><button data-test-id="button">Click me</button><span data-test-id="count">0</span></div></section></body></html><script>window.manifest = {}</script><script type="module" src="/_build/@fs/workspaces/vinxi/test/.fixtures/hmr-dupkt25h3h8/app/entry-client.tsx" async=""></script>

      83 |                      prettyHtml(`<h1 data-test-id="content">Hello from Vinxi too</h1>`),
      84 |              );
    > 85 |      });
         |                   ^
      86 |
      87 |      test("client hmr", async ({ page }) => {
      88 |              let app = new PlaywrightFixture(appFixture, page);

        at file:///workspaces/vinxi/test/hmr.test.ts:85:30


  1 failed
    [chromium] › hmr.test.ts:81:3 › hmr › hmr api ==================================================
  8 passed (35s)
```

Reply:
I will take a look a it again later today with a fresh mind. Looks like more escaping is going on.

```shell
[chromium] › hmr.test.ts:27:3 › hmr › hmr ssr
12:23:33 [vite] Error when evaluating SSR module #vinxi/handler: failed to import "C:UsersDavideDesktop♂inxi    est.fixtureshmr-dnsbukigfu8appentry-server.tsx"
|- Error: Cannot find module 'C:UsersDavideDesktop♂inxi est.fixtureshmr-dnsbukigfu8appentry-server.tsx' imported from 'virtual:'
    at nodeImport (file:///C:/Users/Davide/Desktop/vinxi/node_modules/.pnpm/vite@4.3.9_patch_hash=vtyqdrd2rn7nzuduoqhhzurwua_@types+node@18.17.11/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:54441:25)
    at ssrImport (file:///C:/Users/Davide/Desktop/vinxi/node_modules/.pnpm/vite@4.3.9_patch_hash=vtyqdrd2rn7nzuduoqhhzurwua_@types+node@18.17.11/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:54343:30)
    at eval (virtual:#vinxi/handler:3:50)
    at instantiateModule (file:///C:/Users/Davide/Desktop/vinxi/node_modules/.pnpm/vite@4.3.9_patch_hash=vtyqdrd2rn7nzuduoqhhzurwua_@types+node@18.17.11/node_modules/vite/dist/node/chunks/dep-e8f070e8.js:54405:15)
```

Reply:
Yeah man you are doing amazing work here. This is the biggest overall issue here.. the differences between the OSes when it comes to paths. I can get on a google meet if you want to chat the issues. (my discord username is nksaraf if you want to get in touch there)

Reply:
Thanks for the offer, I am on the solidjs discord as `edivado`. I may send a message your way on Discord in case I get stuck or something is super unclear if that's cool with you. 

Btw the two failed tests are the glob/micromatch failing the `isRoute` check. I am not familiar with `fast-glob` and `micromatch` so am currently looking at their repos and reading up on them.

[Backslash](https://github.com/micromatch/micromatch#backslashes)
>To solve this, you might be inspired to do something like 'foo\\*'.replace(/\\/g, '/'), but this causes another, potentially much more serious, problem. 

*Not elaborating further is great...* 🤣 

After that, it's just the hmr issue that's left. It's a bit of a weird one. Last time I checked file creation was being picked up but not the edits in some cases.